File: blowup.py
'''
----------------------------------------------------------------
-- Created by:     Alireza Teimoori      -----------------------
-- Created on:     Feb 9 2019            -----------------------
-- Created for:    Assignment 1 - blowup -----------------------
-- Course Code:    ICS4U                 -----------------------
-- Teacher Name:   Chris Atkinson        -----------------------
----------------------------------------------------------------
This program a creates a function called "blowUp()"   ----------
This function makes sure that each digit 0-9 that     ----------
appears in the original string is replaced by that    ----------
many occurrences of the character to the right of the digit. ---
----------------------------------------------------------------
'''
def blowUp(input):
    output = "" # Define empty output string
    
    for count in range(len(input)): # Goes through all the characters in input

        charNow = str(input[count]) # finds what character the index is and puts it into "charNow"
        

        if charNow.isdigit(): # Checks if the character is a number
            #print ("dig found") # Notifies that the char is a digit

            try:
                charNext = str(input[count+1]) # for more convenience :D

                if not charNext.isdigit() and charNext != ' ': # If the next character is not a digit and not space
                    #print ("next not dig") # Notifies that the next character is not a digit

                    for counter in range(int(charNow)): # for int(charNow) times, adds the next character to the output
                        #if counter < 1: print (" *detected* ==> " + charNext + " <== replication process begins") # Notifies that the repeating process has started
                        
                        output = output + charNext # It will be added to the output
                        

                    
                elif charNext.isdigit() or charNext == ' ': # If the next character is a digit too, or space, it will just add "charNow" to the output

                    output = output + charNow

            except IndexError:
                #print ("except! hahaha!") # Notifies that an exception has been thrown in the way
                
                
                output = output + charNow # this makes sure that the last character is added to the output as well! :)
                
        else: # If the character is not a digit:

            output = output + charNow # It will be added to the output

    return output

userinput = input("Hello! what do you want to blow up today?! :) \n\n")
print(blowUp(userinput))

File: blowup.java
//-------------------------------------------
//-- Created by:     Alireza Teimoori      --
//-- Created on:     Feb 11 2019           --
//-- Created for:    Assignment 1 - blowup --
//-- Course Code:    ICS4U                 --
//-- Teacher Name:   Chris Atkinson        --
//-------------------------------------------

//import java.util.*;

public class blowup{
    public static String blowUp(String input){
        String output = new String();
        System.out.println("func is started!");

        for (int counter = 0; counter < input.length(); counter ++){ // Goes through all the characters in input
            if (counter < 1){System.out.println("for has started!");} // Notifies that this loop has started

            char charNow = input.charAt(counter); // finds what character the index is and puts it into "charNow"
            char charNext;
            int charValue = Character.getNumericValue(charNow);
            System.out.println(charNow);
            if (Character.isDigit(charNow)){ // Checks if the character is a number
                System.out.println(" --- is digit:" + charNow);
                try {
                    charNext = input.charAt(counter+1);
                    if (Character.isAlphabetic(charNext)){
                        System.out.println(" --- but this is not a digit: " + charNext);
                        for (int count = 0; count < charValue; count++){
                            if (count < 1) {System.out.println("* " + charValue + " reproduction begins: \n");}
                            
                            output = output + charNext;
                            
                        }
                    } else if (Character.isDigit(charNow)){
                        output = output + charNow;
                    }
                } catch (Exception e) {
                    System.out.println("catch! muhahahaha!!!!");
                    output = output + charNow;
                }
            } else if (Character.isAlphabetic(charNow)){
                output = output + charNow;
            }
            
        }
        return output;

    }
    public static void main(String[] args) {
        System.out.println(blowup.blowUp("a3fb4d59"));
    }
        
    
}